src/api/commits.ts

```diff
@@ -0,0 +1,86 @@
+import * as core from '@actions/core'
+import * as github from '@actions/github'
+import { Commit } from '../model/commit'
+import moment from 'moment'
+
+export async function getCommitHistory(
+    token: string,
+    owner: string,
+    repo: string,
+    base: string,
+    head: string): Promise<Commit[]> {
+   return new Promise(async (resolve, reject) => {
+        const octokit = github.getOctokit(token)
+        await octokit.rest.repos.compareCommits({
+            owner: owner,
+            repo: repo,
+            base: base,
+            head: head,
+        }).then((response) => {
+            const commits = response.data.commits.map(c => {
+                return {
+                    author: {
+                        email: c.commit.author?.email ,
+                        name: c.commit.author?.name,
+                        username: c.author?.login
+                    },
+                    message: c.commit.message,
+                    timestamp: c.commit.author?.date,
+                    url: c.commit.url
+                } as Commit
+            })
+            const sortedCommits = sortCommits(commits)
+            resolve(sortedCommits)
+            }).catch(error => {
+                core.warning("Failed to retrieve commits", error);
+                reject(error)
+            });
+    })
+}
+
+export async function getCommitsSinceDate(
+    token: string, 
+    owner: string, 
+    repo: string, 
+    date: string): Promise<Commit[]> {
+    return new Promise(async (resolve, reject) => {
+        const octokit = github.getOctokit(token)
+        await octokit.rest.repos.listCommits({
+            owner: owner,
+            repo: repo,
+            since: date
+        }).then((response) => {
+            const commits = response.data.map(c => {
+                return {
+                    author: {
+                        email: c.commit.author?.email ,
+                        name: c.commit.author?.name,
+                        username: c.author?.login
+                    },
+                    message: c.commit.message,
+                    timestamp: c.commit.author?.date,
+                    url: c.commit.url
+                } as Commit
+            })
```

Comment:
Would it be worth DRYing this up a bit and moving the author generation into its own function?